tests/test_fugue.py

```diff
@@ -268,22 +290,81 @@ def test_doc_case():
     )
 
 
-def test_report_pandas(simple_diff_df1, simple_diff_df2):
+def _compare_report(expected, actual, truncate=False):
+    if truncate:
+        expected = expected.split("Sample Rows", 1)[0]
+        actual = actual.split("Sample Rows", 1)[0]
+    assert expected == actual
+
+
+def test_report_pandas(
+    simple_diff_df1,
+    simple_diff_df2,
+    no_intersection_diff_df1,
+    no_intersection_diff_df2,
+    large_diff_df1,
+    large_diff_df2,
+):
     comp = Compare(simple_diff_df1, simple_diff_df2, join_columns=["aa"])
     a = report(simple_diff_df1, simple_diff_df2, ["aa"])
-    assert a == comp.report()
+    _compare_report(comp.report(), a)
     a = report(simple_diff_df1, simple_diff_df2, "aa", parallelism=2)
-    assert a == comp.report()
+    _compare_report(comp.report(), a)
+
+    comp = Compare(
+        no_intersection_diff_df1, no_intersection_diff_df2, join_columns=["x"]
+    )
+    a = report(no_intersection_diff_df1, no_intersection_diff_df2, ["x"])
+    _compare_report(comp.report(), a)
+    a = report(no_intersection_diff_df1, no_intersection_diff_df2, "x", parallelism=2)
+    _compare_report(comp.report(), a)
+
+    # can't pass due to https://github.com/capitalone/datacompy/issues/221
```

Comment:
given that the issue you flagged is sort of a limitation we can maybe do something like this? appreciate your thoughts.

```python 
@pytest.fixture
def large_diff_df1():
    np.random.seed(0)
    data = np.random.randint(0, 7, size=10000)
    return pd.DataFrame({"x": data, "y": np.array([9]*10000)})


@pytest.fixture
def large_diff_df2():
    np.random.seed(0)
    data = np.random.randint(6, 11, size=10000)
    return pd.DataFrame({"x": data, "y": np.array([9]*10000)})


def test_report_pandas(
    simple_diff_df1,
    simple_diff_df2,
    no_intersection_diff_df1,
    no_intersection_diff_df2,
    large_diff_df1,
    large_diff_df2,
):
    comp = Compare(simple_diff_df1, simple_diff_df2, join_columns=["aa"])
    a = report(simple_diff_df1, simple_diff_df2, ["aa"])
    _compare_report(comp.report(), a)
    a = report(simple_diff_df1, simple_diff_df2, "aa", parallelism=2)
    _compare_report(comp.report(), a)

    comp = Compare(
        no_intersection_diff_df1, no_intersection_diff_df2, join_columns=["x"]
    )
    a = report(no_intersection_diff_df1, no_intersection_diff_df2, ["x"])
    _compare_report(comp.report(), a)
    a = report(no_intersection_diff_df1, no_intersection_diff_df2, "x", parallelism=2)
    _compare_report(comp.report(), a)

    # due to https://github.com/capitalone/datacompy/issues/221
    # we can have y as a constant to ensure all the x matches are equal

    comp = Compare(large_diff_df1, large_diff_df2, join_columns=["x"])
    a = report(large_diff_df1, large_diff_df2, ["x"])
    _compare_report(comp.report(), a, truncate=True)
    a = report(large_diff_df1, large_diff_df2, "x", parallelism=2)
    _compare_report(comp.report(), a, truncate=True)


def test_report_spark(
    spark_session,
    simple_diff_df1,
    simple_diff_df2,
    no_intersection_diff_df1,
    no_intersection_diff_df2,
    large_diff_df1,
    large_diff_df2,
):
    simple_diff_df1.iteritems = simple_diff_df1.items  # pandas 2 compatibility
    simple_diff_df2.iteritems = simple_diff_df2.items  # pandas 2 compatibility
    no_intersection_diff_df1.iteritems = (
        no_intersection_diff_df1.items
    )  # pandas 2 compatibility
    no_intersection_diff_df2.iteritems = (
        no_intersection_diff_df2.items
    )  # pandas 2 compatibility
    large_diff_df1.iteritems = large_diff_df1.items  # pandas 2 compatibility
    large_diff_df2.iteritems = large_diff_df2.items  # pandas 2 compatibility

    df1 = spark_session.createDataFrame(simple_diff_df1)
    df2 = spark_session.createDataFrame(simple_diff_df2)
    comp = Compare(simple_diff_df1, simple_diff_df2, join_columns="aa")
    a = report(df1, df2, ["aa"])
    _compare_report(comp.report(), a)

    df1 = spark_session.createDataFrame(no_intersection_diff_df1)
    df2 = spark_session.createDataFrame(no_intersection_diff_df2)
    comp = Compare(no_intersection_diff_df1, no_intersection_diff_df2, join_columns="x")
    a = report(df1, df2, ["x"])
    _compare_report(comp.report(), a)

    # due to https://github.com/capitalone/datacompy/issues/221
    # we can have y as a constant to ensure all the x matches are equal

    df1 = spark_session.createDataFrame(large_diff_df1)
    df2 = spark_session.createDataFrame(large_diff_df2)
    comp = Compare(large_diff_df1, large_diff_df2, join_columns="x")
    a = report(df1, df2, ["x"])
    _compare_report(comp.report(), a, truncate=True)
```

The key bit being `large_diff_df1` and `large_diff_df2`'s `y` just being some constant so say if there are are a random number of `x` join keys on the first `n` will match in both the fugue and non-fugue version. It should give reproducible results.

Reply:
Thank you so much! I have replaced with your test.